readme.md

This is a program associated with the publication 'Accelerating flux balance calculations in genome-scale metabolic models by localizing the applicaitons of loopless constraints' by Chan SHJ, Wang L, Dash S and Maranas CD (2017).
The files in this folder aim at comparing the presented method with previous methods. For the more general implementation in the COBRA toolbox. Please see the folder 'scripts for COBRA'.


`fastLooplessFVA_mod.m` is the modified implementation of loopless FVA with Fast-SNP preprocessing by Saa and Nielsen, 2016, Bioinformatics.

`localizedLooplessFVA_gurobi.m` was modified from fastLooplessFVA_mod.m to have the same program structure for fair comparison. It supports using only Gurobi as the solver.

`testLooplessFVAmethods.m` performs the test shown in Table 1 and Figure 2 in the paper by calling the two functions described above. Please see the description therein.

`plotFigure2.m` plots the results from the data files generated by testLooplessFVAmethods. Please see the description therein.

All other functions are subroutines called by the first two functions described.

